sqlmesh/core/config/connection.py

```diff
@@ -45,6 +46,11 @@ def _static_connection_kwargs(self) -> t.Dict[str, t.Any]:
         """The static connection kwargs for this connection"""
         return {}
 
+    @property
+    def _execution_config(self) -> t.Optional[BigQueryExecutionConfig]:
```

Comment:
Why does BigQuery-specific config makes it into the base class?

Reply:
Well it is just BigQuery right now but it would be a union of config options as more were created. The optional here means that for non-bq it would return None. It sounds like you prefer a dict so I will be switching to that anyways.